src/revlanguage/distributions/phylogenetics/tree/Dist_FastBirthDeathShiftProcess.cpp

```diff
@@ -0,0 +1,379 @@
+    {
+        cond_num_tips = true;
+    }
+    else if (simulate_cond == "tree")
+    {
+        cond_tree = true;
+    }
+    
+    size_t max_l = static_cast<const Integer &>( max_lineages->getRevObject() ).getValue();
+    size_t min_l = static_cast<const Integer &>( min_lineages->getRevObject() ).getValue();
+    size_t exact_l = static_cast<const Integer &>( exact_lineages->getRevObject() ).getValue();
+    double max_t = static_cast<const RealPos &>( max_time->getRevObject() ).getValue();
+    
+    size_t prune = static_cast<const RlBoolean &>( prune_extinct_lineages->getRevObject() ).getValue();
+    
+    // finally make the distribution 
+    RevBayesCore::FastBirthDeathShiftProcess*   d = new RevBayesCore::FastBirthDeathShiftProcess( 
+            ra, 
+            sp_scale, 
+            ex_scale, 
+            sp_sd, 
+            ex_sd, 
+            r_sp, 
+            r_ext, 
+            num_speciation_classes, 
+            num_extinction_classes, 
+            cond, 
+            uo, 
+            min_l,
+            max_l,
+            exact_l,
+            max_t,
+            prune, 
+            cond_tip_states,
+            cond_num_tips,
+            cond_tree);
+   
+    
+
+    // set sampling probabilities/fractions
+    RevBayesCore::TypedDagNode<double>* rh = static_cast<const Probability &>( rho->getRevObject() ).getDagNode();
+    d->setSamplingFraction( rh );
+    
+   
+    return d;
+}
+
+
+
+/* Get Rev type of object */
+const std::string& Dist_FastBirthDeathShiftProcess::getClassType(void)
+{
+    
+    static std::string rev_type = "Dist_FastBirthDeathShiftProcess";
+    
+    return rev_type;
+}
+
+
+/* Get class type spec describing type of object. TODO: Check if the correct parent is TypedDistribution or Distribution */
+const TypeSpec& Dist_FastBirthDeathShiftProcess::getClassTypeSpec(void)
+{
+    
+    static TypeSpec rev_type_spec = TypeSpec( getClassType(), new TypeSpec( TypedDistribution<TimeTree>::getClassTypeSpec() ) );
+    
+    return rev_type_spec;
+}
+
+
+/**
+ * Get the Rev name for the distribution.
+ * This name is used for the constructor and the distribution functions,
+ * such as the density and random value function
+ *
+ * \return Rev name of constructor function.
+ */
+std::string Dist_FastBirthDeathShiftProcess::getDistributionFunctionName( void ) const
+{
+    // create a distribution name variable that is the same for all instance of this class
+    std::string d_name = "FastBirthDeathShift";
+    
+    return d_name;
+}
+
+
+/**
+ * Get the alternative Rev names (aliases) for the constructor function.
+ *
+ * \return Rev aliases of constructor function.
+ */
+std::vector<std::string> Dist_FastBirthDeathShiftProcess::getDistributionFunctionAliases( void ) const
+{
+    // create alternative constructor function names variable that is the same for all instance of this class
+    std::vector<std::string> a_names;
+    a_names.push_back( "dist_alias_here" );
+
```

Comment:
Don't forget to change this placeholder! Personally, I'd suggest at least "FastBDS" as a valid alias.